pkg/endpointmanager/manager.go

```diff
@@ -130,6 +130,26 @@ func (mgr *endpointManager) WithPeriodicEndpointGC(ctx context.Context, checkHea
 	return mgr
 }
 
+// WithPeriodicEndpointRegeneration configures the EndpointManager to regenerate all (policy and configuration) state
+// of endpoints periodically.
+func (mgr *endpointManager) WithPeriodicEndpointRegeneration(ctx context.Context, interval time.Duration) *endpointManager {
+	mgr.controllers.UpdateController("endpoint-periodic-regeneration",
+		controller.ControllerParams{
+			Group: endpointGCControllerGroup,
+			DoFunc: func(_ context.Context) error {
+				wg := mgr.RegenerateAllEndpoints(&regeneration.ExternalRegenerationMetadata{
+					Reason:            "periodic endpoint regeneration",
+					RegenerationLevel: regeneration.RegenerateWithoutDatapath,
+				})
+				wg.Wait()
```

Comment:
We are calling `mgr.controllers.RemoveAllAndWait()` when the Hive is stopped (which e.g. happens when SIGTERM is received). 

If endpoint regeneration takes potentially a long time to complete, I'm a bit worried that the endpoint manager Hive stop hook blocks because we're waiting on this `wg.Wait()` here, thus preventing other stop hooks (such as the identity allocator checkpointer) from running, because iirc we force-kill cilium-agent if it doesn't stop within a few seconds.

Would it be safe to abort waiting on endpoint regen here if the controller is stopped?

On the other hand, I guess we're using `RegenerateWithoutDatapath` here, so it shouldn't take that long to complete?

Reply:
Good point; I'll stop waiting on the WaitGroup if the context is closed. This will, unfortunately, require another goroutine. But it's only one, not one-per-endpoint.